torch/lib/c10d/ProcessGroupNCCL.cpp

```diff
@@ -140,26 +140,35 @@ void ProcessGroupNCCL::WorkNCCL::wait() {
   synchronize();
 }
 
-std::unordered_map<ssize_t, ssize_t> ProcessGroupNCCL::pgUniqueNCCLIDCnt_;
-ssize_t ProcessGroupNCCL::processGroupCounter_ = -1;
+std::unordered_map<std::string, ssize_t> ProcessGroupNCCL::pgUniqueNCCLIDCnt_;
+std::unordered_map<std::string, ssize_t>
+    ProcessGroupNCCL::processGroupCounterMap_;
+
 std::mutex ProcessGroupNCCL::pgTrackingLock_;
 
 ProcessGroupNCCL::ProcessGroupNCCL(
     const std::shared_ptr<Store>& store,
     int rank,
-    int size)
-    : ProcessGroup(rank, size), store_(store) {
+    int size,
+    const std::string& groupName)
+    : ProcessGroup(rank, size), store_(store), groupName_(groupName) {
   // Generate the Process Group ID for current PG, this needs to be identical
   // for all processes
   std::unique_lock<std::mutex> lock(pgTrackingLock_);
-  ++processGroupCounter_;
-  pgUniqueNCCLIDCnt_[processGroupCounter_] = -1;
-  processGroupID_ = std::to_string(processGroupCounter_);
+  // Default group is an empty string
+  auto groupKey = groupName_ + "_";
+  if (processGroupCounterMap_.count(groupKey) == 0) {
+    processGroupCounterMap_[groupKey] = -1;
+  }
+  ++processGroupCounterMap_[groupKey];
```

Comment:
The default value is 0, so this is equivalent to:

```c++
const auto processGroupCounter = processGroupCounterMap_[groupKey]++;
```

Reply:
Actually, I do prefer writing this way this it's easier to read